3.10/transactions.md

```diff
@@ -1,30 +1,64 @@
 ---
 layout: default
-description: ArangoDB provides support for user-definable  transactions
+description: ArangoDB provides support for user-definable transactions
 ---
-Transactions
-============
+# Transactions
 
-ArangoDB provides support for user-definable transactions.
+## Transaction Types
+
+ArangoDB offers different types of transactions:
+
+- AQL queries (with exceptions)
+- Stream Transactions
+- JavaScript Transactions
+
+<!-- TODO
+### AQL Queries
+
+read own writes (UPSERT?), intermediate commits
+-->
+
+### Stream Transactions
+
+[Stream Transactions](transactions-stream-transactions.html) allow you to perform
+multi-document transactions with individual begin and commit / abort commands.
+They work similar to the *BEGIN*, *COMMIT*, and *ROLLBACK* operations in
+relational database systems.
+
+This API is recommended for larger transactions. However, the client is
+responsible for making sure that transactions are committed or aborted when they
+are no longer needed, to avoid taking up resources.
```

Comment:
Furthermore, we might want to hint to the size limit here and not just say "for large transactions".

Reply:
I copied the text from the HTTP docs, which was probably written by the original author of the feature? I'm not sure what the intention was but it probably refers to these statements that were written about transactions when we only had JavaScript transactions:

> [Transactions are] for short and small data retrieval and/or modification operations.
>
> The implementation is not optimized for very long-running or very voluminous operations […]

Are Stream Transactions really better suited for "larger" transactions than JS transactions, with their size and time limits in mind?

Reply:
We are not sure what the reason for this statement was and can only conjecture that it might refer to the memory limitations and performance overhead of JavaScript transactions because of V8. Removing the statement because it can be misleading - Stream Transactions are not good for large transactions because of the builtin memory limit and timeout.